external/kafka-0-10-token-provider/src/main/scala/org/apache/spark/kafka010/KafkaTokenUtil.scala

```diff
@@ -293,11 +293,11 @@ private[spark] object KafkaTokenUtil extends Logging {
   def isConnectorUsingCurrentToken(
       params: ju.Map[String, Object],
       clusterConfig: Option[KafkaTokenClusterConf]): Boolean = {
-    if (params.containsKey(SaslConfigs.SASL_JAAS_CONFIG)) {
+    if (SparkEnv.get.conf.getBoolean("spark.security.credentials.kafka.enabled", true) &&
```

Comment:
Is there a case where "spark.security.credentials.kafka.enabled" is set to false but "clusterConfig.isDefined" is set to true? I'm wondering whether we only check "clusterConfig.isDefined" and achieve the same thing.

Reply:
> Is there a case where "spark.security.credentials.kafka.enabled" is set to false but "clusterConfig.isDefined" is set to true?

Yes, if one configured the clusters it must be an easy way to turn it off. Actually this is stated in the documentation.

> I'm wondering whether we only check "clusterConfig.isDefined" and achieve the same thing.

It's not the same because my previous comment.

Reply:
OK got it. It would be better if we have sparkConf as a parameter, like other methods in this class. It will help to avoid bothering with NPE issue due to SparkEnv.get being null in test suite, as well as easier to inject Spark configuration during tests.

Reply:
Well, I've considered to add `SparkConf` as parameter but haven't seen neither big advantage nor disadvantage. To make it look like all others seems reasonable + the API is more clear.